File: swift/hardcoded_secrets/PaymentConfig.swift
// PaymentConfig.swift
// Shopist – Payment gateway configuration
// ⚠️  DEMO FILE – Intentionally vulnerable for Datadog Code Security demo
// DO NOT USE IN PRODUCTION

import Foundation

// VULN 1: Hardcoded Stripe secret key
// Live Stripe secret keys must never be committed to source control; any process
// or log that accesses this constant will expose the key.
struct StripeConfig {
    static let stripeKey = "sk_live_4eC39HqLyjWDarjtT1zdp7dc"
    static let stripePublishableKey = "pk_live_TYooMQauvdEDq54NiTphI7jx"
    static let webhookSecret = "whsec_4eC39HqLyjWDarjtT1zdp7dc1234567890"

    static var defaultHeaders: [String: String] {
        return [
            "Authorization": "Bearer \(stripeKey)",
            "Content-Type": "application/x-www-form-urlencoded"
        ]
    }
}

// VULN 2: Hardcoded AWS access key and secret key in a credentials struct
// These credentials grant programmatic access to AWS services (S3, SES, etc.).
// Committing them allows anyone with repo access to enumerate and exfiltrate data.
struct AWSCredentials {
    static let accessKeyId     = "AKIAIOSFODNN7EXAMPLE"
    static let secretAccessKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
    static let region          = "us-east-1"
    static let s3BucketName    = "shopist-product-images"
    static let sesRegion        = "us-east-1"
}

// VULN 3: Hardcoded database password embedded in a connection URL string
// The full connection string (including password) is embedded in source; rotating
// the credential requires a code change and new deployment.
struct DatabaseConfig {
    static let host            = "shopist-prod-db.cluster-ro-abc123.us-east-1.rds.amazonaws.com"
    static let port            = 5432
    static let databaseName    = "shopist_production"
    static let username        = "shopist_admin"
    static let password        = "Sup3rS3cretProdPassw0rd!"
    static let connectionURL   = "postgresql://shopist_admin:Sup3rS3cretProdPassw0rd!@shopist-prod-db.cluster-ro-abc123.us-east-1.rds.amazonaws.com:5432/shopist_production"

    static func buildConnectionString() -> String {
        return "postgresql://\(username):\(password)@\(host):\(port)/\(databaseName)"
    }
}

File: go/hardcoded_secrets/payment_config.go
package hardcodedsecrets

import (
	"database/sql"
	"fmt"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/credentials"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3"
	"github.com/stripe/stripe-go/v74"
	"github.com/stripe/stripe-go/v74/charge"
)

// VULN 1: Hardcoded Stripe secret key - payment charge
func ChargeCustomer(customerID string, amount int64, currency string) (*stripe.Charge, error) {
	stripe.Key = "sk_live_4eC39HqLyjWDarjtT1zdp7dc"
	params := &stripe.ChargeParams{
		Amount:   stripe.Int64(amount),
		Currency: stripe.String(currency),
		Customer: stripe.String(customerID),
	}
	return charge.New(params)
}

// VULN 2: Hardcoded AWS credentials - S3 session for product image uploads
func NewS3Client() *s3.S3 {
	sess := session.Must(session.NewSession(&aws.Config{
		Region: aws.String("us-east-1"),
		Credentials: credentials.NewStaticCredentials(
			"AKIAIOSFODNN7EXAMPLE",
			"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
			"",
		),
	}))
	return s3.New(sess)
}

// VULN 3: Hardcoded DB password in connection string - product database
func OpenProductDB() (*sql.DB, error) {
	connStr := fmt.Sprintf("host=db.shopist.internal port=5432 user=shopist_admin password=Sh0p1st$ecret2024! dbname=shopist_products sslmode=disable")
	return sql.Open("postgres", connStr)
}
